MapboxCoreNavigation/RouteController.swift

```diff
@@ -223,6 +223,12 @@ open class RouteController: NSObject {
     var recentDistancesFromManeuver: [CLLocationDistance] = []
     
     var previousArrivalWaypoint: Waypoint?
+    
+    var userSnapToStepDistanceFromManeuver: CLLocationDistance? {
```

Comment:
Performance-wise, this is no different than before, because this value has to get computed each time the variable is accessed. My suggestion in https://github.com/mapbox/mapbox-navigation-ios/pull/1121#discussion_r168397149 was to compute the distance just once, then either plumb it through to each helper method that needs it or set it on a (stored) property of RouteStepProgress, where the helper methods can access it without any performance penalty.

Reply:
My preference would be to keep the `RouteStep`'s coordinates in a stored `polyline` property, and then have `RouteStep.coordinates` be a computed property that points to the polyline's coordinates.

Reply:
I agree, however the one pitfall with this approach is that it is one more thing to make sure is updated on every progress update.

Reply:
@1ec5 @JThramer fixed in https://github.com/mapbox/mapbox-navigation-ios/pull/1121/commits/14ebaa60d215a94a432bc5c420da89087c8fc452